app/features/worksheet_generator/core.py

```diff
@@ -1,33 +1,40 @@
 from typing import List
 from app.services.logger import setup_logger
-from app.features.worksheet_generator.tools import generate_course_type, worksheet_generator
+from app.features.worksheet_generator.tools import (
+    generate_course_type, 
+    worksheet_generator, 
+    worksheet_question_type_generator
+)
 from app.features.worksheet_generator.document_loaders import get_docs
-from app.services.schemas import WorksheetQuestionModel
 
 logger = setup_logger()
 
 def executor(grade_level: str,
              topic: str,
-             worksheet_list: WorksheetQuestionModel,
              file_url: str,
              file_type: str,
              lang: str,
              verbose=True):
 
-    
     course_type = generate_course_type(topic, verbose)['course_type']
 
     if verbose:
         logger.info(f"Course Type: {course_type} [{lang}]")
         logger.info(f"File URL loaded: {file_url}")
 
     docs = get_docs(file_url, file_type, verbose)
+
+    worksheet_list = worksheet_question_type_generator(course_type=course_type,
```

Comment:
@WilfredoAaronSosaRamosBoushtech this tool has 3 LLM calls with this refactoring now. I was wondering is there a way to combine this in a single call or perhaps you can help me understand your reasoning

Reply:
@yunusj I've done 3 LLM calls following this internal process:
1. The first LLM call is done in order to identify the course type based in the International Baccaulereate guidelines (For adapting the LLM understanding of the nature, methodology and body of knowledge of the course types):
https://www.ibo.org/programmes/diploma-programme/curriculum/
2. The second LLM call is to identify what question types and number of questions are required for the given inputs.
3. The third LLM call is for generating the worksheet.

Reply:
I used multiple shot prompting for adapting the generated Worksheet in base of the nature and methodology of the course type based in the International Baccaulereate.